com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -387,14 +399,31 @@ internal NetworkObject CreateLocalNetworkObject(bool isSceneObject, uint globalO
                         return null;
                     }
 
-                    // Otherwise, instantiate an instance of the NetworkPrefab linked to the prefabHash
-                    var networkObject = ((position == null && rotation == null) ? UnityEngine.Object.Instantiate(networkPrefabReference) : UnityEngine.Object.Instantiate(networkPrefabReference, position.GetValueOrDefault(Vector3.zero), rotation.GetValueOrDefault(Quaternion.identity))).GetComponent<NetworkObject>();
-
+                    // Create prefab instance
+                    // Note: We used to set position here, see comments below.
+                    var networkObject = UnityEngine.Object.Instantiate(networkPrefabReference).GetComponent<NetworkObject>();
                     networkObject.NetworkManagerOwner = NetworkManager;
 
+                    // Go ahead and set network parenting properties
+                    networkObject.SetNetworkParenting(isReparented, parentNetworkId, worldPositionStays);
+
+                    if (sceneObject.Header.HasTransform)
+                    {
+                        if (worldPositionStays)
+                        {
+                            networkObject.transform.position = position;
+                            networkObject.transform.rotation = rotation;
+                        }
+                        else
+                        {
+                            networkObject.transform.localPosition = position;
+                            networkObject.transform.localRotation = rotation;
+                        }
+                    }
+
                     if (parentNetworkObject != null)
                     {
-                        networkObject.transform.SetParent(parentNetworkObject.transform, true);
+                        networkObject.TrySetParent(parentNetworkObject.transform, worldPositionStays);
                     }
```

Comment:
nit: you're doing this 2 times, maybe 3 with prefabhandler above... feels like a good candidate for a private method?

Reply:
Are you talking about lines 402 - 427 and lines 439-472?

Reply:
@JesseOlmer 
I consolidated this code.